flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/ContinuousEventTimeTrigger.java

```diff
@@ -45,6 +45,12 @@
 	private final ReducingStateDescriptor<Long> stateDesc =
 			new ReducingStateDescriptor<>("fire-time", new Min(), LongSerializer.INSTANCE);
 
+	/**
+	 * Used to preserve the fire timestamp before merge such that
+	 * the corresponding timer could be cleared after merge
+	 */
+	private Long cachedFireTimestamp = null;
```

Comment:
This doesn't work because there is only one `Trigger` instance and this will potentially be overwritten by many Windows being merged.